datasets/flwr_datasets/metrics/utils_test.py

```diff
@@ -19,12 +19,126 @@
+        ),
+        (
+            datasets.Dataset.from_dict(
+                {"feature": list(range(10)), "label": [0, 0, 0, 1, 1] + [1, 1, 1, 1, 2]}
+            ),
+            pd.DataFrame(
+                [[3, 2, 0], [0, 4, 1]], index=pd.Index([0, 1], name="Partition ID")
+            ),
+        ),
+    ],
+)
+class TestPublicMetricsUtils(unittest.TestCase):
+    """Test metrics utils."""
+
+    dataset: datasets.Dataset
+    result: pd.DataFrame
+
+    def test_compute_counts(self) -> None:
+        """Test if the counts are computed correctly."""
+        iid_partitioner = IidPartitioner(num_partitions=2)
+        iid_partitioner.dataset = self.dataset
+        count = compute_counts(iid_partitioner, column_name="label")
+        pd.testing.assert_frame_equal(count, self.result)
+
+    def test_compute_frequencies(self) -> None:
+        """Test if the frequencies are computed correctly."""
+        print(self.dataset.features)
+        iid_partitioner = IidPartitioner(num_partitions=2)
+        iid_partitioner.dataset = self.dataset
+        frequencies = compute_frequencies(iid_partitioner, column_name="label")
+        result = self.result.div(self.result.sum(axis=1), axis=0)
+        pd.testing.assert_frame_equal(frequencies, result)
+
+    def test_compute_counts_with_verbose_label(self) -> None:
+        """Test if the counts are computed correctly."""
+        iid_partitioner = IidPartitioner(num_partitions=2)
+        dataset = self.dataset
+        new_col_names = [
+            str(col_id) for col_id in range(len(self.dataset.unique("label")))
+        ]
+        dataset = dataset.cast_column(
+            "label",
+            ClassLabel(
+                num_classes=len(self.dataset.unique("label")), names=new_col_names
+            ),
+        )
+        iid_partitioner.dataset = dataset
+        result = self.result.copy()
+        result.columns = new_col_names
+        count = compute_counts(iid_partitioner, column_name="label", verbose_names=True)
+        pd.testing.assert_frame_equal(count, result)
+
+    def test_compute_frequencies_with_verbose_label(self) -> None:
+        """Test if the frequencies are computed correctly."""
+        iid_partitioner = IidPartitioner(num_partitions=2)
+        dataset = self.dataset
+        new_col_names = [
+            str(col_id) for col_id in range(len(self.dataset.unique("label")))
+        ]
+        dataset = dataset.cast_column(
+            "label",
+            ClassLabel(
+                num_classes=len(self.dataset.unique("label")), names=new_col_names
+            ),
+        )
+        iid_partitioner.dataset = dataset
+        result = self.result.copy()
+        result.columns = new_col_names
+        result = result.div(result.sum(axis=1), axis=0)
+        frequencies = compute_frequencies(
+            iid_partitioner, column_name="label", verbose_names=True
+        )
+        print(frequencies)
```

Comment:
remove?

Reply:
Yes, sorry, I was debugging and forgot about it